### ISSUE TITLE
【iOS】Keyboard events' deal cause screen confusion

### ISSUE THREAD

**Issue by qiyer** [cocos2d/cocos2d-x#17746]
- cocos2d-x version: all version (3.2 3.7 3.9 test)
- devices test on: iphone 6 ,iphone 6 P,iphone 6s ,iphone 6s P,iphone 7 ,iphone 7 P
- developing environments
   - Xcode version:Version 8.2.1 (8C1002)

Steps to Reproduce:
 
1. The project is landscape . (iOS project Device Orientation setting supports Landscape & Portrait)
2. The project should have an EditBox on the screen.
3. run the project on device，then switch screen .
4. you can open wechat or QQ ，chat just show the keyboard.
5. back the project，great probability cause screen confusion.

操蛋，我还是直接说中文得了。2dx 在初始化 一个EditBox 输入框后，就一直在侦听键盘事件，侦听就算了，你还一直处理 你们的GLView 大小和位置。有时候对接第三方SDK ，第三方有输入框 键盘事件变化你们也处理，切到微信、QQ等聊天 键盘你们也处理。有的第三方可是有强制竖屏的界面，这时候只要 横屏游戏配置了 竖屏支持，就有很大概率 出现 错屏 现象。

花了半天时间才定位到2dx问题。解决方案就是 修改 CCEAGLView-ios.mm （老版本是CCEAGLView.mm）。
修改
-(void) doAnimationWhenKeyboardMoveWithDuration:(float)duration distance:(float)dis; 函数实现switch 的第一个case：
        case UIInterfaceOrientationPortrait:
            if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0)
            {
                self.frame = CGRectMake(originalRect_.origin.x, originalRect_.origin.y - dis, self.frame.size.width, self.frame.size.height);
            }else{
                self.frame = CGRectMake(originalRect_.origin.x, originalRect_.origin.y - dis, originalRect_.size.width, originalRect_.size.height);
            }
            break;  
改好，经过多机型多轮测试 fix。稍后我会 pull 一个过来。

**Comment by minggo** [comment]
So should stop listening keyboard event instead of change the size?

**Comment by qiyer** [comment]
Yes, i try to stop listening keyboard event, but it will effect 2dx‘s Editbox .Then, i just change the size. The best to , stop listening keyboard event for the right time.

**Comment by qiyer** [comment]
最好是在合适的时机 移除键盘监听， 我尝试过 这样做法，但是会影响 Editbox 的键盘伸缩上下位置。
最后我只好修改了size，能解决问题，但不是最优的方案。
我之前有在你们Editbox监听事件回调里面判断 当前事件是否是 Editbox传来的，根据编辑状态判断，但是会影响Editbox，所以不能以编辑状态作为判断依据，最好你们能加入一个新的标示，比如tag 之类的，判断一下。
（因为我不是cocos2d-x 开发者，是一名 iOS SDK开发，对你们引擎不是很熟悉，所以也没深入 研究。）

**Comment by minggo** [comment]
thanks
